dts/bindings/base/pm.yaml

```diff
@@ -16,6 +16,11 @@ properties:
       Wake up capable devices are disabled (interruptions will not wake up
       the system) by default but they can be enabled at runtime if necessary.
 
+  wakeup-source-ids:
+    type: array
+    description: |
+      Property to identify vendor specific wakeup sources.
```

Comment:
On STM32, there is a complete decorrelation between wakeup pin numbering and GPIO. This is an information that you need to get from the datasheet. For instance:
![wkup_2](https://github.com/zephyrproject-rtos/zephyr/assets/18574308/e3e400e8-aa66-43fb-b987-db67cb7d541c)
Then control is done using the wakeup pin numbering through dedicated registers (non related to GPIO)
![wkup_pwr_reg](https://github.com/zephyrproject-rtos/zephyr/assets/18574308/a8a36fb3-3dbf-44fb-8c9a-3930bb56a0ad)
This is typically the thing useful to get from device tree rather than hardcoding a table to be updated for each and other STM32 variant.

Can this property be used to convey this piece of information ?

Reply:
Yep :) pretty much exactly what it's for

Reply:
Not really, device pins are used in the dts overlays like this
```
sensor: bmi323@0 {
	compatible = "bosch,bmi323"
	reg = <0>;
	gpios = <&gpio0 11 (GPIO_ACTIVE_HIGH)>;
	wakeup-source;
}
```
so the user would have to look up the mapping and apply it manually, it is intended for peripherals defined in the .dtsi, like an RTC for example.
```
rtc: rtc@4000000 {
	compatible = "atmel,sam-rtc";
	reg = <4000000 0x100>;
	interrupts = <2 0>,
	             <3 0>;
	interrupt-names = "update", "wakeup";
	alarms-count = <1>;
	status = "disabled";
};
```

Not really, the GPIOs are defined like this

Reply:
> Not really, device pins are used in the dts overlays like this

Ok, got it.

> The SOC will have to implement int z_sys_wakeup_source_enable_gpio(const struct gpio_dt_spec *gpio) and keep some internal header or similar to map the port + pin to the wakeup number

Then, if we can't use `wakeup-source-ids`, we should be able to find a way to use device tree to describe this mapping, right ?